libs/server/Lua/LuaCommands.cs

```diff
@@ -0,0 +1,227 @@
+        /// EVALSHA sha1 numkeys [key [key ...]] [arg [arg ...]]
+        /// </summary>
+        /// <param name="count"></param>
+        /// <returns></returns>
+        private unsafe bool TryEVALSHA(int count)
+        {
+            if (!storeWrapper.serverOptions.EnableLua)
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_LUA_DISABLED, ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            var digest = parseState.GetArgSliceByRef(0).ReadOnlySpan;
+
+            var result = false;
+            if (!sessionScriptCache.TryGetFromDigest(digest, out var runner))
+            {
+                var d = digest.ToArray();
+                if (storeWrapper.storeScriptCache.TryGetValue(d, out var source))
+                {
+                    if (!sessionScriptCache.TryLoad(source, d, out runner, out var error))
+                    {
+                        while (!RespWriteUtils.WriteError(error, ref dcurr, dend))
+                            SendAndReset();
+                        _ = storeWrapper.storeScriptCache.TryRemove(d, out _);
+                        return result;
+                    }
+                }
+            }
+            if (runner == null)
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_NO_SCRIPT, ref dcurr, dend))
+                    SendAndReset();
+            }
+            else
+            {
+                result = ExecuteScript(count - 1, runner);
+            }
+            return result;
+        }
+
+
+        /// <summary>
+        /// EVAL script numkeys [key [key ...]] [arg [arg ...]]
+        /// </summary>
+        /// <param name="count"></param>
+        /// <returns></returns>
+        private unsafe bool TryEVAL(int count)
+        {
+            if (!storeWrapper.serverOptions.EnableLua)
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_LUA_DISABLED, ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            var script = parseState.GetArgSliceByRef(0).ReadOnlySpan;
+            var digest = sessionScriptCache.GetScriptDigest(script);
+
+            var result = false;
+            if (!sessionScriptCache.TryLoad(script, digest, out var runner, out var error))
+            {
+                while (!RespWriteUtils.WriteError(error, ref dcurr, dend))
+                    SendAndReset();
+                return result;
+            }
+            if (runner == null)
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_NO_SCRIPT, ref dcurr, dend))
+                    SendAndReset();
+            }
+            else
+            {
+                result = ExecuteScript(count - 1, runner);
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// SCRIPT Commands (load, exists, flush, kills)
```

Comment:
KILL is not implemented; maybe add a TODO case for it

Reply:
removed KILL from comment, can be added in future.